### PR TITLE
[Speedy] Move back contract cache in speedy

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
In order to reactive tracing of contract lookup in the EvaluationOrderTest

### CODE REVIEW

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala [hunk 1]
```diff
@@ -404,7 +401,7 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
                 { granted: Boolean =>
                   if (granted) {
                     callback()
-                    loopOuter(cache)
```
Comment (basvangijzel-DA) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala:407]: This was previously:
interpretLoop(machine, time)

Any reason for changing this back differently?

**Review by basvangijzel-DA — COMMENTED** [review]
For reference, this undoes some of the changes from https://github.com/digital-asset/daml/pull/17171.

Is it worth adding back in the test Nick ignored? 
https://github.com/digital-asset/daml/commit/4b60e35a5d7fc31362381edcbe04a3dcea7a0e02#diff-fa63d1a40eb0bdb9ec5809e1447d4ebfcb3f8a160df7d739dac43a2c1f87140f

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala:407]: a07f8f1d2e1d04b416733fcf322b80928ccdfb57 reverts back the previous version

**Review by basvangijzel-DA — APPROVED** [review]
Thanks Rémy!

**Pull request merged**